src/App/Repository/Form/StatsRepository.php

```diff
@@ -279,4 +350,99 @@ private function targetedSurveyStateJoin($query)
             ->join('posts', 'INNER')
             ->on('posts.id', '=', 'targeted_survey_state.post_id');
     }
+
+    public function getSurveyType($form_id)
+    {
+        $query = DB::select('targeted_survey')
+        ->from('forms')
+        ->where('id', '=', $form_id);
+
+        $results = $query->execute($this->db);
+        return $results->as_array();
+    }
+
+    public function getPostCountByDataSource($form_id, $created_after, $created_before)
+    {
+        if ($created_after) {
+            $created_after = strtotime($created_after);
+        }
+
+        if ($created_before) {
+            $created_before = strtotime($created_before);
+        }
+
+        $result = [
+            'sms' => $this->queryByDataSource($form_id, $created_after, $created_before, 'sms'),
+            'email' => $this->queryByDataSource($form_id, $created_after, $created_before, 'email'),
+            'twitter' => $this->queryByDataSource($form_id, $created_after, $created_before, 'twitter'),
+            'web' => $this->queryForWeb($form_id, $created_after, $created_before),
+            'all' => $this->queryForAllPosts($form_id, $created_after, $created_before)
+        ];
+
+        return $result;
+    }
+
+    private function queryByDataSource($form_id, $created_after, $created_before, $source)
+    {
+        $query = DB::select([DB::expr('COUNT(messages.id)'), 'total'])
```

Comment:
I just realized that we could do a GROUP_BY('messages.type') and return all the datasource counts in 1 query /faster. LMK if you want to do that, we can pair a bit and do it. Of course this only applies to the queryByDatasource calls